runtime/src/lib.rs

```diff
@@ -158,7 +158,7 @@ pub trait Sink: Sync + Send + 'static {
 /// Interface that any runtime must implement to receive
 /// messages over a network connection.
 pub trait Stream: Sync + Send + 'static {
-    fn recv(&mut self) -> impl Future<Output = Result<Bytes, Error>> + Send;
+    fn recv(&mut self, buf: &mut[u8]) -> impl Future<Output = Result<usize, Error>> + Send;
```

Comment:
To mirror the interface for `Storage`, we should just return `Result<(), Error>` here.

Reply:
We should clarify this here and in `Storage` (that we wait for the buffer to be full before returning).